examples/tracingpolicy/cves/cve-2024-21626-runc.yaml

```diff
@@ -0,0 +1,134 @@
+# image is set up to execute /proc/self/fd/7/../../../bin/bash and run a shell
+# script that overwrites /proc/self/exe, overwriting the host copy of /bin/bash.
+# The CVSS score for this attack is CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:C/C:H/I:H/A:H (8.6, high severity).
+#
+# Attack 3b is attack 2 but adapted to overwrite a host binary, where the malicious
+# container process overwrites all of the possible runc exec target binaries inside
+# the container (such as /bin/bash) such that a host target binary is executed and
+# then the container process opens /proc/$pid/exe to get access to the host binary
+# and overwrite it.
+# The CVSS score for this attack is CVSS:3.1/AV:L/AC:L/PR:L/UI:R/S:C/C:H/I:H/A:H (8.2, high severity).
+#
+# This Tracing Policy does not protect against Attacks 3a and 3b.
+#
+#
+# This cve-2024-21626-runc Tetragon Tracing Policy can be used to detect usage of
+# magic symlinks /proc/* that are passed to runc in order to change current working
+# directory. The detection is performed at the system call layer.
+#
+# Further, this Tracing Policy checks at kernel layer if the current working
+# directory of current container process is being set to '/sys/fs/cgroup' which is
+# one of the leaked file descriptors in runc that was fixed in:
+# https://github.com/opencontainers/runc/commit/506552a88bd3455e80a9b3829568e94ec0160309
+#
+# While this Tracing Policy is able to block '/sys/fs/cgroup' file descriptors leaks and
+# block CVE-2024-21626 exploits that change the working directory, it is not intended as
+# a complete prevention of all possible file descriptors leaks and exploits for runc.
+# If there are more leaks from the host filesystem, then adding those resolved host
+# filesystem paths to the 'set_fs_pwd' kprobe section, under the 'matchArgs' to detect
+# usage of such leaks will certainly block all of them.
+#
+
+apiVersion: cilium.io/v1alpha1
+kind: TracingPolicy
+metadata:
+  name: "cve-2024-21626-runc"
+  annotations:
+    url: "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-21626"
+    description: "Prevents exploitation of runc CVE-2024-21626"
+    author: "Tetragon.io Team"
+spec:
+  kprobes:
+  - call: "sys_chdir"
+    syscall: true
+    message: "Possible exploit of runc cve-2024-21626 vulnerability"
+    args:
+      - index: 0
+        type: "string"
+    selectors:
+    - matchBinaries:
+      - operator: "In"
+        values:
+        # Add your runc paths here
+        - "/usr/bin/runc"
+        - "/usr/sbin/runc"
+        - "/usr/local/sbin/runc"
+      matchArgs:
+      - index: 0
+        operator: "Prefix"
+        values:
+        - "/proc/self/fd/"
+      matchActions:
+        - action: Sigkill
+  - call: "set_fs_pwd"
+    syscall: false
+    message: "Possible exploit of runc cve-2024-21626 vulnerability"
+    args:
+      - index: 1
+        type: "path"
+    selectors:
+    - matchBinaries:
+      - operator: "In"
+        values:
+        # Add your runc paths here
```

Comment:
is there a requirement to specificy the path to runc? it seems trivial to move the binary out of the path and still execute?